pkg/api/resourcegroup/v1alpha1/types.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package defines ResourceGroup schema.
+// Version: v1alpha1
+// swagger:meta
+package v1alpha1
+
+import (
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+const (
+	RGFileName       = "resourcegroup.yaml"
+	RGFileKind       = "ResourceGroup"
+	RGFileGroup      = "kpt.dev"
+	RGFileVersion    = "v1alpha1"
+	RGFileAPIVersion = RGFileGroup + "/" + RGFileVersion
+)
+
+// TypeMeta is the TypeMeta for ResourceGroup instances.
+var TypeMeta = yaml.ResourceMeta{
+	TypeMeta: yaml.TypeMeta{
+		APIVersion: RGFileAPIVersion,
+		Kind:       RGFileKind,
+	},
+}
+
+// ResourceGroup contains the inventory information about a package managed with kpt.
+// swagger:model resourcegroup
```

Comment:
just curious, what is the swagger:model needed for?

Reply:
The swagger marker is used to generate the [swagger schema](https://github.com/GoogleContainerTools/kpt/blob/3d90e0ed19bcc5759dc22105aec46e53a0e15715/Makefile#L47). This is mainly for use/reference by [kpt.dev](https://kpt.dev) under the references/schema section.

Reply:
Got it. thanks.

Reply:
sg.